src/test/TestUtils.cpp

```diff
@@ -194,6 +197,78 @@ genesis(int minute, int second)
         getTestDateTime(1, 7, 2014, 0, minute, second));
 }
 
+void
+upgradeSorobanNetworkConfig(std::function<void(SorobanNetworkConfig&)> modifyFn,
+                            SimulationPtr simulation)
+{
+    auto nodes = simulation->getNodes();
+    auto& lg = nodes[0]->getLoadGenerator();
+    auto& app = *nodes[0];
+
+    auto& complete =
+        app.getMetrics().NewMeter({"loadgen", "run", "complete"}, "run");
+    auto completeCount = complete.count();
+    // Only create an account if there are none aleady created.
+    uint32_t offset = 0;
+    if (app.getMetrics()
+            .NewMeter({"loadgen", "account", "created"}, "account")
+            .count() == 0)
+    {
+        auto createAccountsLoadConfig =
+            GeneratedLoadConfig::createAccountsLoad(1, 1);
+        offset = std::numeric_limits<uint32_t>::max() - 1;
+        createAccountsLoadConfig.offset = offset;
+
+        lg.generateLoad(createAccountsLoadConfig);
+        simulation->crankUntil(
+            [&]() { return complete.count() == completeCount + 1; },
+            300 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+    }
+
+    // Create upload wasm transaction.
+    auto createUploadCfg = GeneratedLoadConfig::createSorobanUpgradeSetupLoad();
+    createUploadCfg.offset = offset;
+    lg.generateLoad(createUploadCfg);
+    completeCount = complete.count();
+    simulation->crankUntil(
+        [&]() { return complete.count() == completeCount + 1; },
+        300 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
+    // Create upgrade transaction.
+    auto createUpgradeLoadGenConfig = GeneratedLoadConfig::txLoad(
+        LoadGenMode::SOROBAN_CREATE_UPGRADE, 1, 1, 1);
+    createUpgradeLoadGenConfig.offset = offset;
+    // Get current network config.
+    auto cfg = nodes[0]->getLedgerManager().getSorobanNetworkConfig();
+    modifyFn(cfg);
+    createUpgradeLoadGenConfig.copySorobanNetworkConfigToUpgradeConfig(cfg);
+    auto upgradeSetKey = lg.getConfigUpgradeSetKey(createUpgradeLoadGenConfig);
+    lg.generateLoad(createUpgradeLoadGenConfig);
+    completeCount = complete.count();
+    simulation->crankUntil(
+        [&]() { return complete.count() == completeCount + 1; },
+        300 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
+    // Arm for upgrade.
+    for (auto app : nodes)
+    {
+        Upgrades::UpgradeParameters scheduledUpgrades;
+        auto lclHeader =
+            app->getLedgerManager().getLastClosedLedgerHeader().header;
+        scheduledUpgrades.mUpgradeTime =
+            VirtualClock::from_time_t(lclHeader.scpValue.closeTime);
+        scheduledUpgrades.mConfigUpgradeSetKey = upgradeSetKey;
+        app->getHerder().setUpgrades(scheduledUpgrades);
+    }
+    // Wait for upgrade to be applied
+    simulation->crankUntil(
+        [&]() {
+            auto netCfg = app.getLedgerManager().getSorobanNetworkConfig();
+            return netCfg == cfg;
+        },
+        300 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
```

Comment:
300 ledgers seems like a really long timeout, in reality you should see the upgrade going through on the next ledgers, so you can set it to 2 to be safe.